controllers/k8ssandra/stargate.go

```diff
@@ -192,3 +198,104 @@ func (r *K8ssandraClusterReconciler) removeStargateStatus(kc *api.K8ssandraClust
 		}
 	}
 }
+
+func (r *K8ssandraClusterReconciler) reconcileStargateConfigMap(
+	ctx context.Context,
+	remoteClient client.Client,
+	kc *api.K8ssandraCluster,
+	desiredConfig map[string]interface{},
+	userConfigMapContent,
+	namespace string,
+	logger logr.Logger,
+) result.ReconcileResult {
+	logger.Info("Reconciling Stargate Cassandra yaml configMap on namespace : " + namespace)
+	filteredCassandraConfig := filterYamlConfig(desiredConfig["cassandra-yaml"].(map[string]interface{}))
+
+	configYamlString := ""
+	if len(filteredCassandraConfig) > 0 {
+		if configYaml, err := yaml.Marshal(filteredCassandraConfig); err != nil {
+			return result.Error(err)
+		} else {
+			configYamlString = string(configYaml)
+		}
+	}
+
+	if userConfigMapContent != "" {
+		separator := "\n"
+		if strings.HasSuffix(userConfigMapContent, "\n") {
+			separator = ""
+		}
+		configYamlString = userConfigMapContent + separator + configYamlString
```

Comment:
What if the user configmap contains encryption entries? I think this could result in an invalid yaml file that will cause Stargate to fail to start.

I think it would be safer to manipulate maps instead:

```go
finalConfigMapContent := make(map[string]interface{}, 0)
for k, v := range userConfigMapContent { // assuming userConfigMapContent is a map not a string
	finalConfigMapContent[k] = v
}
for k, v := range filteredCassandraConfig {
	finalConfigMapContent[k] = v
}
finalConfigYaml, err := yaml.Marshal(finalConfigMapContent)
```

Reply:
Also, the current implementation might have a problem in the case where the user does not supply any configmap: I think in this case the YAML string that is generated here will contain only the encryption settings and nothing else, which is not what we want.

Reply:
> in this case the YAML string that is generated here will contain only the encryption settings and nothing else, which is not what we want.

I just confirmed with @jsanda that this is not a problem, so please disregard my comment.